### PR TITLE
Remove "oauth2" scheme from openapi security scheme

### LINKED ISSUE #4071: [OpenApi] Additional property "scheme" in SecuritySchema
**API Platform version(s) affected**: 2.6.2

**Description**  

```
  securitySchemes:
    oauth:
      scheme: oauth2 <--- this one
```

The error is : components.securitySchemes.oauth should NOT have additional properties. This field is also not mentioned on https://swagger.io/docs/specification/authentication/oauth2/

**How to reproduce**  
Enable the implicit flow in the apiplatform configuration.

**Possible Solution**  
The value is nullable but its pre set here 
https://github.com/api-platform/core/blob/959f0829747eb733f496b5ddc6508b433b4f6792/src/OpenApi/Factory/OpenApiFactory.php#L459

Set the default value to null.

Edit: I just saw that its used for BasicAuth.
```
components:
  securitySchemes:
    basicAuth: 
      type: http
      scheme: basic
```

### PR DESCRIPTION
**Pull request by WhiteRabbitDE**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| Deprecations? | no
| Tickets       | Fix #4071
| License       | MIT

### CODE REVIEW

**Review by alanpoulain — APPROVED** [review]
The `scheme` key seems only to be used for `Basic` and `Bearer` indeed.
LGTM but @soyuka needs to confirm.

**Comment by alanpoulain** [comment]
Please fix the unit tests too.

**Review by soyuka — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 * Mercure: Do not use data in options when deleting (#4056)
 * Doctrine: Support for foreign identifiers
 * SchemaFactory: Allow generating documentation when property and method start from "is" (property `isActive` and method `isActive`)
+* OpenApiFactory: Do not set scheme to oauth2 when generating securitySchemes
```
Comment (alanpoulain) [inline CHANGELOG.md:9]:
```suggestion
* OpenAPI: Do not set scheme to oauth2 when generating securitySchemes (#4073)
```

**Comment by alanpoulain** [comment]
Thank you @WhiteRabbitDE.

**Pull request merged**